java/clients/src/main/java/sleeper/clients/deploy/properties/GeneratePropertiesDocumentation.java

```diff
@@ -0,0 +1,132 @@
+                        stream -> writePropertiesMarkdownFile(new TableProperties(new InstanceProperties()), tablePropertyGroup, stream));
+            } catch (Exception e) {
+                System.out.println("Unable to write property file for group: " + tablePropertyGroup.getName());
+            }
+        });
+    }
+
+    private static void generateMasterPage(Path root) {
+        try {
+            writeFile(root.resolve("docs/usage/property-master.md"),
+                    stream -> generateDocumentLinks(stream));
+        } catch (Exception e) {
+            System.out.println("Unable to generate property master document");
+        }
+    }
+
+    private static void generateDocumentLinks(OutputStream stream) {
+        try {
+            PrintStream out = new PrintStream(stream, true, StandardCharsets.UTF_8.name());
+            out.println("## Instance Properties");
+            out.println();
+            InstancePropertyGroup.getAllGroupNamesAlphabetically().forEach(group -> {
+                out.println(pageLinkFromGroupName(group, "instance/"));
+            });
+            out.println();
+            out.println("## Table Properties");
+            out.println();
+            TablePropertyGroup.getAllGroupNamesAlphabetically().forEach(group -> {
+                out.println(pageLinkFromGroupName(group, "table/"));
+            });
+        } catch (UnsupportedEncodingException e) {
+            System.out.println("Unable to encode output file");
+        }
+    }
+
+    private static String pageLinkFromGroupName(String group, String directory) {
+        return String.format("[%s](properties/%s%s)<br>", group, directory, groupNameToFileName(group));
+    }
+
+    /**
+     * Generates table containing all the properties for a given property group in markdown format.
+     * This is then written out to a file named for the property group.
+     *
+     * @param  group                        the group of properties
+     * @param  out                          the stream for the output
+     * @throws UnsupportedEncodingException thrown if uft8 unavailable
+     */
+    @SuppressWarnings({"rawtypes", "unchecked"})
+    private static void writePropertiesMarkdownFile(SleeperProperties properties, PropertyGroup group, OutputStream out) {
+        try {
+            PrintStream stream = new PrintStream(out, true, StandardCharsets.UTF_8.name());
+            stream.println("## " + group.getName().toUpperCase(Locale.ENGLISH));
+            stream.println();
+            stream.println("Below is a table containing all the details for the property group: " + group.getName());
+            stream.println();
+            TableWriter tableWriter = SleeperPropertyMarkdownTable.generateTableBuildForGroup(properties.getPropertiesIndex().getAllInGroup(group).stream());
+            tableWriter.write(stream);
+        } catch (UnsupportedEncodingException e) {
+            System.out.println("Unable to encode output file");
```

Comment:
It looks important to fail the build if an exception happens here, because it means the file didn't write properly. It looks like the exception should be thrown all the way out?